iChef-WebApi/src/main/java/com/kitchen/iChef/Domain/AppUser.java

```diff
@@ -0,0 +1,30 @@
+package com.kitchen.iChef.Domain;
+
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+@Entity
+@Data
+public class AppUser {
+    @Id
+    private String userId;
+    private String firstName;
+    private String lastName;
+    private String username;
+    private String email;
+    private ZonedDateTime joinedDate;
+    private ZonedDateTime lastOnline;
+    private Boolean isAdmin;
+    private String hashedPassword;
+
+    @PrePersist
+    private void ensureId() {
```

Comment:
Nitpick: rename to generate Id? No strong feelings

Reply:
is a premade name of the method used by the Generator, no need to change the method name